### ISSUE TITLE
CLUSTER BY and OPTIMIZE unparseable in the databricks dialect

### ISSUE THREAD

**Issue by pgrz** [sqlfluff/sqlfluff#5145]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm using databricks 3.0 with liquid clustering.

`USING DELTA CLUSTER BY (columns)` and `OPTIMIZE table` are reported as unparseable.

### Expected Behaviour

These should be correctly recognized as dialect keywords as per https://docs.databricks.com/en/delta/clustering.html

### Observed Behaviour

`Found unparsable section: '\nCLUSTER BY (somecolumn)'`

`Found unparsable section: 'OPTIMIZE sometable'`

### How to reproduce

Try to parse the following:
```sql
CREATE TABLE tablename
(
    id_column INT,
    othercolumn STRING
)
USING DELTA
LOCATION "s3://someplace"
CLUSTER BY (id_column);

OPTIMIZE tablename;
```

### Dialect

databricks

### Version

2.3.1

### Configuration

pyproject.toml file:
```toml
[tool.sqlfluff.core]
dialect = "databricks"
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
